rholang/project/BNFC.scala

```diff
@@ -50,8 +51,11 @@ object BNFC {
     bnfcOutputDir  := (javaSource in Compile).value,
     bnfcDocDir     := baseDirectory.value / "doc" / "bnfc",
     clean          := cleanDir(bnfcOutputDir.value / nsToPath(bnfcNamespace.value)),
-    generate       := bnfcFiles(bnfcGrammarDir.value).foreach { (f: File) =>
-      bnfcGenerateSources((fullClasspath in BNFCConfig).value, f, bnfcOutputDir.value, bnfcNamespace.value)
+    generate       := {
+      val fullCP = (fullClasspath in BNFCConfig).value
+      bnfcFiles(bnfcGrammarDir.value).foreach { (f: File) =>
+        bnfcGenerateSources(fullCP, f, bnfcOutputDir.value, bnfcNamespace.value)
+      }
```

Comment:
likewise, is this needed to get the code to compile with scala 2.12.4?

Reply:
This is another requirement from sbt 1.x for task invocation inside anonymous functions.